File: ticket_granting_service/kerberos_tgs/time_validator.py
class TimeValidator:
    TGT_EXPIRATION_TIME_FORMAT = '%d/%m/%Y %H:%M'
    
    REQUESTED_TIME_FORMAT = '%d/%m/%Y %H:%M'

    @classmethod
    def tgt_expiration_time_is_valid(cls, tgt_expiration_time_str):
        """Verifica se o prazo de validade do TGT está no formato certo"""

        try:
            expiration_date = datetime.strptime(tgt_expiration_time_str, cls.TGT_EXPIRATION_TIME_FORMAT)
            return True
        except ValueError:
            return False

    @classmethod
    def requested_time_is_valid(cls, requested_time_str):
        """Verifica se o tempo requisitado está em um dos formatos válidos"""

        # O TGS implementado só trabalha com um formato de "tempo
        # de acesso": "dd/mm/yyyy HH:MM", que significa que o
        # cliente quer ou está autorizado a acessar o serviço até
        # essa data.
        # 
        # Caso fosse interessante, ele poderia ser aprimorado
        # para aceitar outras formas de os clientes especificarem
        # um tempo de acesso:
        #   Ex: Em um período específico --> "FROM 10:00 TO 22:00"
        #       Somente alguns dias --> "DAYS=[mon, tue, wed, thu, fri]"
        try:
            requested_time = datetime.strptime(requested_time_str, cls.REQUESTED_TIME_FORMAT)
            return True
        except ValueError:
            return False

File: service1/timeValidator.js

const moment = require('moment')

const REQUESTED_TIME_FORMAT = 'DD/MM/YYYY HH:mm'
const AUTORIZED_TIME_FORMAT = 'DD/MM/YYYY HH:mm'

/**
 * Verifica se o tempo solicitado está no formato certo.
 * @param {string} requestedTimeStr Tempo solicitado
 */
function requestedTimeIsValid(requestedTimeStr) {
    var requestedTime = moment(requestedTimeStr, REQUESTED_TIME_FORMAT, true)

    return requestedTime.isValid()
}

/**
 * Verifica se o tempo autorizado está em um formato válido.
 * @param {string} autorizedTimeStr Tempo autorizado
 */
function autorizedTimeIsValid(autorizedTimeStr) {
    /* O TGS implementado só trabalha com um formato de "tempo
    de acesso": "dd/mm/yyyy HH:MM", que significa que o
    cliente quer ou está autorizado a acessar o serviço até
    essa data.
        Se tivessem outros formatos (ex: um período específico,
    somente alguns dias da semana...), o serviço precisaria
    verificar se é algum destes formatos. */

    var autorizedTime = moment(autorizedTimeStr, AUTORIZED_TIME_FORMAT, true)

    return autorizedTime.isValid()
}

/**
 * Verifica se o tempo solicitado pelo cliente está
 * dentro do tempo autorizado pelo TGS.
 * @param {string} requestedTimeStr Tempo solicitado pelo cliente
 * @param {string} autorizedTimeStr Tempo autorizado pelo TGS
 */
function isAuthorized(requestedTimeStr, autorizedTimeStr) {
    var requestedTime = moment(requestedTimeStr, REQUESTED_TIME_FORMAT, true).toDate()
    var autorizedTime = moment(autorizedTimeStr, AUTORIZED_TIME_FORMAT, true).toDate()

    if (requestedTime < autorizedTime) {
        return true
    } else {
        return false
    }
}

module.exports = {
    requestedTimeIsValid: requestedTimeIsValid,
    autorizedTimeIsValid: autorizedTimeIsValid,
    isAuthorized: isAuthorized
}